apps/emqx/src/emqx_persistent_session_ds/emqx_persistent_session_ds_stream_scheduler.erl

```diff
@@ -1210,23 +1216,29 @@ runtime_state_invariants(ModelState, #{s := S, stream_scheduler_s := SchedS}) ->
 offline_state_invariants(_ModelState, #{s := S}) ->
     invariant_active_streams_y_ranks(S).
 
-%% Verify that each active topic subscription has a new stream watch:
+%% Verify that each active durable topic subscription has a new stream watch:
 invariant_new_stream_subscriptions(#{subs := Subs}, #s{
     new_stream_subs = Watches, sub_metadata = SubStates
 }) ->
-    ExpectedTopics = lists:sort(maps:keys(Subs)),
+    ExpectedTopics = lists:sort(
+        [
+            Topic
+         || {Topic, SubOpts = #{qos := QoS}} <- maps:to_list(Subs),
+            QoS > ?QOS_0 orelse maps:get(durable, SubOpts, false)
```

Comment:
I think this invariant can be changed to "for each model subscription there should be *either* a direct or durable broker subscription". This should eliminate the need for `durable` flag in the model.